### ISSUE TITLE
TextAlignment and BoundingBoxes

### ISSUE THREAD

**Issue by sschmidTU** [opensheetmusicdisplay/opensheetmusicdisplay#355]
There seems to be a general issue for alignments of GraphicalLabels and how we deal with their position and coordinates. We currently do not modify the position of the GraphicalLabel's bounding box, but simply shift the screen coordinates that we got from the label's original position. This example is not from me, though i followed it for left-alignment:
```javascript
const screenPosition: PointF2D = this.applyScreenTransformation(graphicalLabel.PositionAndShape.AbsolutePosition);
(...)
case TextAlignmentEnum.CenterTop:
                screenPosition.x -= <number>bitmapWidth / 2;
```
from MusicSheetdrawer.ts/drawLabel()

For center alignment, this seems to work out for the bounding boxes, though i'm not sure why.
![image](https://user-images.githubusercontent.com/33069673/44810946-9f11e780-abd3-11e8-99b3-2dd21d152782.png)

For left-alignment, in order for lyrics to appear left-aligned, i had to shift them by about 1 unit to the left (our coordinates, 10 pixels in Vexflow).
```javascript
case TextAlignmentEnum.LeftBottom:
  screenPosition.y -= bitmapHeight;
  // without the following, lyrics are not optically left-aligned. influences lyrics, dash positioning.
  // TODO unify alignment shifts and our bbox position, which does not correspond to screenposition
  screenPosition.x -= unitInPixels;
```

This means that our label bounding boxes do not fit what's on the screen for left-alignment:
![image](https://user-images.githubusercontent.com/33069673/44809593-6f60e080-abcf-11e8-883f-9e65261f5f8f.png)

I'll try to fix this by shifting the label's position instead of the screenPosition. I just didn't figure out how to do that yet. The RelativePosition of the label gets overwritten at some point (tried setting it to x=-1), and i'm not sure i should set the AbsolutePosition in the layout calculation.

**Comment by sschmidTU** [comment]
I changed back the placement of non-lyrics (LeftTop, LeftCenter) to how they were before (c13432c03e6b4d685f2b4eda870e1e0bb3a5b3c3). It affected the placement of the Instrument name and the lyricist (see Land der Berge), and i'm not sure that's intended.
This is because lyrics were not optically left-aligned with the settings we also used for labels like instrument name. For now, LeftBottom (lyrics) gets more x-shift (-1) again.

Compare images of [state of develop and most recent branch commit](https://user-images.githubusercontent.com/33069673/44856317-658dba80-ac6d-11e8-95dd-ca6bcb808602.png)
and [equal handling of all left-aligned shifts, which i reverted for now](https://user-images.githubusercontent.com/33069673/44856336-750d0380-ac6d-11e8-84db-a27a3794db67.png).

It's also a question of where the position (x in the image) should be: should the bounding box extend only to the right (first image), or a bit to the left for left-aligned labels (second image)?

**Comment by sschmidTU** [comment]
Talking about it with matt-uib, we agreed that we should set the Label's (Relative)Position correctly instead of manipulating the screenPosition too much.
The screenPosition should only be manipulated for discrepances between our bounding box position and the VexFlow draw position.

So, x=0 for left-aligned lyrics should be at the start of the text, like in [this picture](https://user-images.githubusercontent.com/33069673/44809593-6f60e080-abcf-11e8-883f-9e65261f5f8f.png), just with the correct bounding box.

I'll try to set the position correctly again. I tried setting the RelativePosition.x to -1, but it got overwritten at a later point.

**Comment by sschmidTU** [comment]
Left-aligned lyrics bbox is fixed in 4c837944d819c78b4269235dbe89ca525d81c452.

The "general issue" with alignments, bounding boxes and screen positions i opened the issue for is resolved for me now.
The screenPosition shifts in MusicSheetdrawer.ts/drawLabel() are adjustments for the TextAlignments: For a LeftBottom alignment, the position of the BoundingBox is in the left bottom corner, and extends to the top and right direction.

The only remaining issue is that these are also adjustments for VexFlow screen positions which perhaps should be in VexFlowMusicSheetDrawer and not MusicSheetDrawer, but i added a TODO for that in the same commit.
